lib/puppet/provider/package/rpm.rb

```diff
@@ -281,6 +281,7 @@ def self.nevra_to_hash(line)
       self::NEVRA_FIELDS.zip(match.captures) { |f, v| hash[f] = v }
       hash[:provider] = self.name
       hash[:ensure] = "#{hash[:version]}-#{hash[:release]}"
+      hash[:ensure].prepend("#{hash[:epoch]}:") if hash[:epoch] != '0'
```

Comment:
The rpm provider uses [`Puppet::Util::Package.versioncmp`](https://github.com/whopper/puppet/blob/PUP-5025/RPM-epoch/lib/puppet/provider/package/rpm.rb#L142) to compare ensure values, but that method doesn't know how to handle epochs in version strings, such as what we're returning here.

Do we want to update `versioncmp` to handle epochs?

Or implement an `rpmversioncmp` that knows how to handle epochs? Based on http://www.rpm.org/max-rpm-snapshot/s1-rpm-depend-manual-dependencies.html:

>  Epoch is the most significant component of a package's complete version identifier with regards to RPM's version comparison algorithm.

I assume we'd want to first compare epochs, defaulting to epoch 0 if one is not specified. If the epochs are different, then return +/-1. If the epochs are the same, then delegate to the original `versioncmp`?

Assuming epochs are specific to rpms, I think we'd want to do the latter. Thoughts?